### ISSUE TITLE
[BUG] Testing react components. Strange reaction

### ISSUE THREAD

**Issue by mikeogi** [microsoft/playwright#22931]
### System info
- Playwright Version: `"@playwright/experimental-ct-react" - 1.33.0`
- Operating System: macOS 13.3.1 ventura

### Source code

- [ ] I provided exact source code that allows reproducing the issue locally.

**Button.spec.tsx**
Failed case
```ts
import { test, expect } from '@playwright/experimental-ct-react';

import Button from './Button.component';

test('render the base button', async ({ mount }) => {
  const component = await mount(<Button>click me</Button>);
  await expect(component).toHaveScreenshot();
});

test.use({ viewport: { width: 1000, height: 900 } });

test('render the full width button', async ({ mount }) => {
  const component = await mount(<Button fullWidth>click me</Button>);
  await expect(component).toHaveScreenshot();
});

// test('test text', async ({ mount }) => {
//   const component = await mount(<span>click me</span>);
//   await expect(component).toContainText('click me');
// });
```

**Button.spec.tsx**
Success case
```ts
import { test, expect } from '@playwright/experimental-ct-react';

import Button from './Button.component';

test('render the base button', async ({ mount }) => {
  const component = await mount(<Button>click me</Button>);
  await expect(component).toHaveScreenshot();
});

test.use({ viewport: { width: 1000, height: 900 } });

test('render the full width button', async ({ mount }) => {
  const component = await mount(<Button fullWidth>click me</Button>);
  await expect(component).toHaveScreenshot();
});

```

**Steps**
- write the test
- add commented test below

**Expected**
the comment does not affect the work

**Actual**
```bash
Error: Cannot find module '/Users/mikeglyc/job/wallet/ui-kit/src/components/Button/Button.component' imported from /Users/mikeglyc/job/wallet/ui-kit/src/components/Button/Button.spec.tsx
```

**Comment by mikeogi** [comment]
test breaks on any change. The only suspicion is that may be is saved by the local cache playwright? I tried to completely remove it from the project and put. Only previously written tests work and without a single change

**Comment by peterkrieg** [comment]
I'm seeing some weird caching behavior too with playwright component tests for react, same version (`1.33.0`).  For me, removing the cache (`rm -rf ./playwright/.cache`) doesn't help, there seems to be some other caching done on the file path level.

@mikeogi  just checking, if you make a one-letter change to your directory where your code is, do the tests pass as expected?  This seems to "bust" the cache.  I'm wondering if playwright team can share some guidance on what might be cached on this file-level, and how to clear this, until a fix is deployed.  

I'm also pretty sure that this is only happening after upgrading from version `1.31.2` =>  `1.32.0`.  The diff between those versions is quite large, I was trying to see what might be causing it.

**Comment by MaxLeiter** [comment]
>I'm also pretty sure that this is only happening after upgrading from version 1.31.2 => 1.32.0. The diff between those versions is quite large, I was trying to see what might be causing it.

I'm seeing

```Error: Cannot find module '/name.pw-test-components' imported from name.pw-test.tsx```

only in 1.33, not 1.31 or 1.32.

**Comment by mikeogi** [comment]
@peterkrieg, thx. It's helped me, but problem in 1.33 exists

**Comment by pavelfeldman** [comment]
We'll get this fixed in 1.34.

The problem is that VS Code extension is running `npx playwright test --list` which does not account for the component testing plugins.

As a workaround, locate your temporary folder:

```sh
find /var/folders | grep playwright-transform
```

and delete all the folders that have `playwright-transform-cache` in the name.

**Comment by mikeogi** [comment]
@pavelfeldman thank you. It helped me, but i stay on version 1.32, she is more stable

**Comment by peterkrieg** [comment]
Thank you @pavelfeldman  for the workaround here - this works for me as a temporary fix.  Looking forward to `1.34.0`.  You guys are crushing it with the quick turnaround on bugs!  🙌 

> We'll get this fixed in 1.34.
> 
> The problem is that VS Code extension is running `npx playwright test --list` which does not account for the component testing plugins.
> 
> As a workaround, locate your temporary folder:
> 
> ```shell
> find /var/folders | grep playwright-transform
> ```
> 
> and delete all the folders that have `playwright-transform-cache` in the name.
